crates/block-producer/Cargo.toml

```diff
@@ -24,7 +24,7 @@ itertools             = { workspace = true }
 miden-block-prover    = { git = "https://github.com/0xPolygonMiden/miden-base.git", branch = "next" }
 miden-lib             = { workspace = true }
 miden-node-proto      = { workspace = true }
-miden-node-utils      = { workspace = true }
+miden-node-utils      = { workspace = true, features = ["testing"] }
```

Comment:
The `testing` feature should only be set for `development` and not in the general case. 

Though I suggest we move all the test code out into a separate PR, i.e. remove the feature entirely for now.